control-plane-operator/controllers/hostedcontrolplane/cvo/reconcile.go

```diff
@@ -377,22 +388,26 @@ func cvoBootstrapScript(clusterVersionJSON []byte) string {
 	payloadDir := volumeMounts.Path(cvoContainerBootstrap().Name, cvoVolumePayload().Name)
 	scriptTemplate := `#!/bin/bash
 set -euo pipefail
+MANIFEST_DIR=%s/manifests
+ls -la ${MANIFEST_DIR}
 cat > /tmp/clusterversion.json <<EOF
 %s
 EOF
 oc get ns openshift-config &> /dev/null || oc create ns openshift-config
 oc get ns openshift-config-managed &> /dev/null || oc create ns openshift-config-managed
+oc apply -f ${MANIFEST_DIR}/0000_00_cluster-version-operator_01_clusterversions*
```

Comment:
why do we need this additional commands now?

Reply:
Some of it was added to help troubleshooting - I'll remove those lines.

This line (398) specifically was added here because we needed to create the ClusterVersion resource earlier than the rest of the stuff in the manifests dir, so that the ClusterVersion was created before the CVO itself, since it's not possible to edit the field after the CVO is running (i.e on a running cluster).
It was @tjungblu who unearthed this requirement, but if I remember correctly the reasoning was that the [resource definition itself did not yet exist](https://github.com/openshift/hypershift/pull/5315#discussion_r1895786134) at this point, and since we wanted to create the ClusterVersion before the CVO, we had to move the CRD creation up here.

Maybe I should add a code comment here to clarify this?

Reply:
feel free to revert them, but it took me three weeks to debug a yaml identation issue with this silly EOF bash imputation. And then it took me another two weeks to figure out why the apiserver was rejecting the apply and didn't complain about it. 

So I think it's useful, but I don't know how often you guys want to touch that code anyway.

Reply:
Thanks for all that work, Thomas, I'm planning on leaving the parsing separated from the apply, just like you ended up doing. The troubleshooting I was referring to I will remove is the `ls` and `oc get ... -oyaml` lines